src/Sarif.Viewer.VisualStudio.Core/Models/SarifErrorListItem.cs

```diff
@@ -113,6 +114,52 @@ public SarifErrorListItem(Run run, int runIndex, Result result, string logFilePa
             }
         }
 
+        /// <summary>
+        ///  Gets the queries that can be used to do codefinding for this error list item.
+        /// </summary>
+        /// <returns>A list of queries for each location. If a location does not require a query, it will be inserted as null. If this item does not require a query for any, this list will be null or empty.</returns>
+        public List<(Uri filePath, MatchQuery query)?> GetMatchQueries()
```

Comment:
The list is always there, its just that the tuples inside the list can be null, which happens if there is not enough information to make a MatchQuery object for a particular location in the SarifResult.Locations array.
I prefer this because it allows for doing snippet matching even if not every location in the array has all of the information we require to match